pkg/apis/extensions/v1beta1/conversion.go

```diff
@@ -93,6 +96,70 @@ func Convert_v1_PodSpec_To_api_PodSpec(in *v1.PodSpec, out *api.PodSpec, s conve
 	return v1.Convert_v1_PodSpec_To_api_PodSpec(in, out, s)
 }
 
+func convertToMapSelector(selector *string) (map[string]string, error) {
+	p := labels.NewParser(labels.NewLexer(*selector, 0))
+	reqs, err := p.Parse()
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse the selector: %v", err)
+	}
+	mapSelector := make(map[string]string)
+	for _, req := range reqs {
+		if req.Operator() != labels.EqualsOperator {
+			return nil, fmt.Errorf("only map-based requirements can be converted")
+		}
+		vals := req.Values()
+		if vals.Len() != 1 {
+			return nil, fmt.Errorf("equals operator must have exactly one value")
+		}
+		val, ok := vals.PopAny()
+		if !ok {
+			return nil, fmt.Errorf("equals operator must have exactly one value")
```

Comment:
It would be useful for this to return a different message than the one above, because this really shouldn't happen.